base_request.py

```diff
@@ -93,35 +88,30 @@ def __request(self, url, method, params, endpoint, headers=None, data=None, stre
     def request(self, url, method, endpoint, params=None, data=None, stream=None, auth=True):
         if auth:
             if not self.token.has():
-                # TODO: need exception if no token provided
-                pass
+                raise ResinException.ResinNotLoggedIn()
 
             if self.util.should_update_token(self.token.get_age(),self.settings.get('token_refresh_interval')):
                 self.token.set(self.request_new_token())
-
         params = params or {}
         data = data or {}
         # About response obj: https://github.com/kennethreitz/requests/blob/master/requests/models.py#L525
         response = self.__request(url, method, params, endpoint, data=data, stream=stream, auth=auth)
-        # FOR TESTING ONLY
-        #print response.status_code
-        #print response.reason
-        #print response.content
-        #print response.url
-        #
+
         if response.status_code == 201:
             return response.content
         # 204: no content
         if response.status_code == 204:
             return
+
+        if stream:
```

Comment:
request with stream=true only for downloading device Os and I only see the response with 200. Anw, I've updated as you suggest to make sure it works